teamcity/tests/test_teamcity.py

```diff
@@ -137,3 +121,104 @@ def test_config(test_case, extra_config, expected_http_kwargs):
         http_wargs.update(expected_http_kwargs)
 
         r.assert_called_with(ANY, **http_wargs)
+
+
+@pytest.mark.parametrize(
+    'build_config, expected_error',
+    [
+        pytest.param(
+            {'projects': {'project_id': {}}}, 'Failed to establish a new connection', id="One `projects` config"
+        ),
+        pytest.param(
+            {'build_configuration': 'build_config_id'},
+            'Failed to establish a new connection',
+            id="One `build_configurations` config",
+        ),
+        pytest.param(
+            {'projects': {'project_id': {}}, 'build_configuration': 'build_config_id'},
+            'Only one of `projects` or `build_configuration` may be configured, not both.',
+            id="Redundant configs",
+        ),
+    ],
+)
+def test_validate_config(dd_run_check, build_config, expected_error, caplog):
+    """
+    Test that the `build_configuration` config options are properly configured in Python 2 prior to running check.
+    Note: The properly configured test cases would be expected to have a `Failed to establish a new connection`
+    exception.
+    """
+    caplog.clear()
+    config = {'server': 'server.name', 'use_openmetrics': False}
+
+    instance = deepcopy(config)
+    instance.update(build_config)
+
+    check = TeamCityRest('teamcity', {}, [instance])
+
+    with pytest.raises(Exception) as e:
```

Comment:
You've probably guessed part of my feedback by now:

1. can we assert about a specific error type at this point? It could be multiple ones as long as they are more specific than `Exception`.
2. I think you can save some code (lines 161 to 167) by using the `match` parameter of [pytest raises](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.raises) eg

```python
with pytest.raises(Exception, match=expected_error):  # you no longer need bind the exception object to a variable
    dd_run_check(check)
```